consensus/enclave/impl/src/lib.rs

```diff
@@ -581,26 +570,50 @@ impl ConsensusEnclave for SgxConsensusEnclave {
     }
 }
 
-/// Creates a single output belonging to the fee recipient account.
+/// Creates a single output belonging to a specific recipient account.
+/// The output is created using a predictable private key that is derived from the input parameters.
 ///
 /// # Arguments:
-/// * `tx_private_key` - Transaction key used to output the aggregate fee.
-/// * `total_fee` - The sum of all fees in the block.
-fn mint_aggregate_fee(
-    fee_recipient: &PublicAddress,
-    tx_private_key: &RistrettoPrivate,
-    total_fee: u64,
+/// * `recipient` - The recipient of the output.
+/// * `domain_tag` - Domain separator for hashing the input parameters.
+/// * `parent_block` - The parent block.
+/// * `transactions` - The transactions that are included in the current block.
+/// * `amount` - Output amount.
+fn mint_output<T: Digestible>(
+    recipient: &PublicAddress,
+    domain_tag: &'static [u8],
+    parent_block: &Block,
+    transactions: &[T],
+    amount: u64,
 ) -> Result<TxOut> {
+    // Create a determinstic private key based on the block contents.
+    let tx_private_key = {
+        let mut hash_value = [0u8; 32];
+        {
+            let mut transcript = MerlinTranscript::new(domain_tag);
+            parent_block
+                .id
+                .append_to_transcript(b"parent_block_id", &mut transcript);
+            transactions.append_to_transcript(b"transactions", &mut transcript);
+            transcript.extract_digest(&mut hash_value);
+        };
```

Comment:
my main comment is, ideally we would also hash the recipient and amount in as well

Reply:
That happens in `MintTx` since they will be part of that struct.